src/Page/Form.purs

```diff
@@ -177,53 +180,80 @@ component =
         (HE.input $ DropdownQuery field.key)
     Text input ->
       Input.input
-        [ HP.value $ fromMaybe "" $ toString =<< getValue input
+        [ HP.value $ Data.Maybe.fromMaybe "" $ toString =<< getValue input
         , HP.placeholder $ print input.placeholder
         , HP.id_ field.key
+        , HE.onValueChange (HE.input $ UpdateValue field.key <<< UserInput <<< Val <<< String)
         ]
     Toggle input ->
       Toggle.toggle
-        [ HP.checked $ fromMaybe false $ toBoolean =<< getValue input
+        [ HP.checked $ Data.Maybe.fromMaybe false $ toBoolean =<< getValue input
```

Comment:
Feels weird to have a default here for when no default is specified. I feel like that means we would have `getValue toggle == Nothing` but on the form it would appear to be `false`. Maybe we need support in Ocelot for a toggle that isn't set. A `Maybe Boolean` toggle perhaps 🤔

Reply:
I think this is an artifact of both halogen stuff taking `Array _` as arguments and there being a sensible default for `checked`. It's easier to handle the `Maybe Boolean` inside of the array than it is to handle it on the outside. What we should probably do is decide whether to add the `checked` property based on what `getValue >=> toBoolean` returns. We _should_ be doing this in all places where we use `getValue`.

In other words, I think `ocelot` is fine to keep doing what it's doing. We're not using it the best way.

Reply:
Hmm, that's true, but there's still the issue that even if we don't pass in `checked` at all, the toggle block might not have a render state that represents that. I think it would just treat it as unchecked or `false`

Reply:
Is that not expected?

Reply:
Maybe? It's kind of weird to have `required` at all on there then. Because if it's set to `true` and no `default` is defined, and the user doesn't ever toggle it to on, they're going to get a very unexpected `Required` error